src/_pytest/_code/code.py

```diff
@@ -549,7 +549,7 @@ def getrepr(
         funcargs: bool = False,
         truncate_locals: bool = True,
         chain: bool = True,
-    ):
+    ) -> Union["ReprExceptionInfo", "ExceptionChainRepr"]:
```

Comment:
Functions returning a Union are not great, because the caller must disambiguate which one they got.

In this case, both types share a common base class `ExceptionRepr`. If all the caller cares about is getting an `ExceptionRepr` from this function, I suggest using that instead. This will also be more future proof.

If however the distinction is important (not in this case AFAIU), I think it will be better to either use two overloads on `style`, `Literal["long"]` which only returns `ExceptionChainRepr` and  `Literal["native"]` which only returns `ReprExceptionInfo`. Or just split the function to two.

Reply:
Thanks for the input.
I've tried to be as strict / verbose as possible, given the code around all this.
Will consider using `ExceptionRepr` then.

Reply:
Not sure if you decided to keep it this way.

Other than this, LGTM.